### PR TITLE
feat: tested to detect linked accounts for code

### PR DESCRIPTION
**Pull request by markusahlstrand**
Patched the code flow so it uses handles linked users. It's a bit tricky as we're actually fetching the user and store it in a the state, so there's not a single point where we can handle the linked accounts. Maybe it would be better to keep the state leaner and try to have one point where we fetch the profiles.

### CODE REVIEW

File: src/helpers/account-linking.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import { Env, Profile } from "../types";
+
+export async function handleLinkedAccount(env: Env, profile: Profile) {
```
Comment (markusahlstrand) [inline src/helpers/account-linking.ts:3]: We can't do this in the DO as it's, with the current setup, not possible to fetch one User DO from within another DO. Feels like a code smell..

File: src/routes/tsoa/authenticate.ts [hunk 2]
```diff
@@ -92,12 +89,20 @@ export class AuthenticateController extends Controller {
       const coVerifier = randomString(32);
       const coID = randomString(12);
 
+      const profile = await handleLinkedAccount(
```
Comment (markusahlstrand) [inline src/routes/tsoa/authenticate.ts:92]: We need to fetch the profile in here and store it in the state as it's were the next endpoint will pick it up. Not fetching the profile here before was probably an issue for generating the id-tokens correctly..

File: test/models/User.spec.ts [hunk 3]
```diff
@@ -491,7 +491,7 @@ describe("User", () => {
 
       expect(profile.connections.length).toBe(1);
       expect(profile.connections[0]).toEqual({
-        name: "linked-user",
+        name: "linked-user|user2@example.com",
```
Comment (danielgent) [inline test/models/User.spec.ts:494]: Cool ok. This shows it works then

Comment (danielgent) [reply test/models/User.spec.ts:494]: Hmmmm, does it? :smile: 

Would be cool to have a test that tries to log in with a linked user...

**Review by danielgent — APPROVED** [review]

**Pull request merged**